### PR TITLE
 Prepare democracy and multisig to be published with aragon-cli

### PR DESCRIPTION
**Pull request by izqui**
This makes production kits ready to be published with the Aragon CLI.

I have made changed to aragen as well, that can now deploy these two kits: https://github.com/aragon/aragen/pull/15/commits/296b310f6e3cdab5c8625922bfdb4335e4df287c

### CODE REVIEW

File: kits/multisig/package.json [hunk 1]
```diff
@@ -18,7 +18,9 @@
     "coverage": "./node_modules/@aragon/test-helpers/run-coverage.sh",
     "console": "truffle dev",
     "ganache-cli:dev": "./node_modules/@aragon/test-helpers/ganache-cli.sh",
-    "ganache-cli:coverage": "SOLIDITY_COVERAGE=true npm run ganache-cli:dev"
+    "ganache-cli:coverage": "SOLIDITY_COVERAGE=true npm run ganache-cli:dev",
+    "publish:rpc": "aragon apm publish $(npm run deploy:kit:rpc | tail -n 1) --network rpc",
+    "deploy:kit:rpc": "aragon contracts compile --network rpc && aragon contracts exec --network rpc scripts/multisig-deploy.js"
```
Comment (sohkai) [inline kits/multisig/package.json]: It seems really weird we need to use `--network rpc` for `aragon contracts compile` but I'm guessing the CLI requires it?

File: kits/multisig/contracts/MultisigKit.sol [hunk 2]
```diff
@@ -58,8 +75,8 @@ contract MultisigKit is BetaKitBase {
         uint256 multisigSupport = neededSignatures * 10 ** 18 / signers.length - 1;
         voting.initialize(
             token,
-            multisigSupport,
-            multisigSupport,
+            uint64(multisigSupport),
+            uint64(multisigSupport),
```
Comment (sohkai) [inline kits/multisig/contracts/MultisigKit.sol]: Could we use the [`Uint256Helpers`](https://github.com/aragon/aragonOS/blob/dev/contracts/common/Uint256Helpers.sol) here and in the democracy kit?

File: kits/multisig/contracts/MultisigKit.sol [hunk 3]
```diff
@@ -14,7 +14,24 @@ contract MultisigKit is BetaKitBase {
         BetaKitBase(_fac, _ens, _minimeFac, _aragonID, _appIds) public
     {}
 
-    function newToken(string name, string symbol) external returns (MiniMeToken token) {
+    function newDemoInstance(string name) external {
+        address[] memory holders = new address[](1);
```
Comment (bingen) [inline kits/multisig/contracts/MultisigKit.sol]: A Multisig 1 out of 1 is kind of funny ;-)

File: kits/beta-base/scripts/deploy_kit.js [hunk 4]
```diff
@@ -40,36 +46,52 @@ module.exports = async (
...
   const kit = await artifacts.require(kitName).new(daoFactory.address, ens.address, minimeFac.address, aragonid, appIds)
 
+  if (returnKit) {
+    return kit
+  }
+
   const ts = [ { name: kitName, address: kit.address } ]
   log(ts)
 
+  log('Creating APM package with owner', owner)
```
Comment (sohkai) [inline kits/beta-base/scripts/deploy_kit.js]: I think this only happens if the network is `devnet`?

File: kits/democracy/contracts/DemocracyKit.sol [hunk 5]
```diff
@@ -15,7 +15,44 @@ contract DemocracyKit is BetaKitBase {
         public
     {}
 
-    function newToken(string name, string symbol) external returns (MiniMeToken token) {
+    function newDemoInstance(string name) external {
```
Comment (sohkai) [inline kits/democracy/contracts/DemocracyKit.sol]: Not a super big fan of having this available by default; this seems like it'd be better as a "demo config" that's specified and parsed off-chain by the CLI.

File: kits/beta-base/scripts/deploy_kit.js [hunk 6]
```diff
@@ -40,36 +46,52 @@ module.exports = async (
...
-    log("Local testing network, exiting...")
-    return;
-  }
-
   let indexFileName
-  if (network != 'rpc' && network != 'devnet') {
-    indexFileName = 'index.js'
-  } else {
-    indexFileName = 'index_local.js'
+  if (!returnKit) {
```
Comment (bingen) [inline kits/beta-base/scripts/deploy_kit.js]: If `returnKit` is false and no ENS address is provided by param or in `ENV`, `indexFileName` below in line 58 will be undefined.

File: kits/democracy/scripts/democracy-deploy.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const deployKit = require('@aragon/kits-beta-base/scripts/deploy_kit.js')
+
+module.exports = async (callback) => {
+  const deployConfig = {
```
Comment (bingen) [inline kits/democracy/scripts/democracy-deploy.js]: Just curious, where are you defining `ENS` for `deployKit` when you run `deploy:kit:rpc`? `ensAddress` nor `network` are being passed, so I understand it should be in `process.ENV`, but I don't see where it's set.

File: kits/multisig/scripts/multisig-deploy.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+const deployKit = require('@aragon/kits-beta-base/scripts/deploy_kit.js')
+
+module.exports = async (callback) => {
+  const deployConfig = {
```
Comment (bingen) [inline kits/multisig/scripts/multisig-deploy.js]: Same question as with democracy.

File: kits/multisig/package.json (hunk 1 above)
Comment (izqui) [reply kits/multisig/package.json]: Yup, agreed. It is a bug in the CLI which requires passing a network for that command.

Comment (izqui) [reply kits/multisig/package.json]: Will be fixed here: https://github.com/aragon/aragon-cli/pull/230/files#r227244994

File: kits/democracy/contracts/DemocracyKit.sol (hunk 5 above)
Comment (izqui) [reply kits/democracy/contracts/DemocracyKit.sol]: Good point, let's remove it for now and figure out how to do this well in the future.

File: kits/beta-base/scripts/deploy_kit.js (hunk 6 above)
Comment (izqui) [reply kits/beta-base/scripts/deploy_kit.js]: That's true, adding a explicit error.

File: kits/democracy/scripts/democracy-deploy.js (hunk 7 above)
Comment (izqui) [reply kits/democracy/scripts/democracy-deploy.js]: Yes, it assumes `ENS` has been passed as an environment variable. In the case of aragen it gets exported here: https://github.com/aragon/aragen/blob/863cd750d8d5c326c624ebee13362d72b316b406/scripts/set-env

**Comment by izqui** [comment]
Address most comments, waiting for the release candidate versions to be cut to update dependencies here, and test again.

**Pull request merged**